### PR TITLE
Add CPython 3.10 wheels

### PR DESCRIPTION
**Pull request by hrnciar**
## PR Summary
This PR modifies `cibuildwheel.yml` to create wheels for Python 3.10. I propose this as a draft because it is currently blocked on numpy's wheel (https://github.com/numpy/numpy/issues/19630).

## PR Checklist

- [N/A] Has pytest style unit tests (and `pytest` passes).
- [N/A] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [N/A] New features are documented, with examples if plot related.
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [N/A] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: .github/workflows/cibuildwheel.yml [hunk 1]
```diff
@@ -64,6 +64,17 @@ jobs:
         run: |
           cp setup.cfg.template setup.cfg
 
+      - name: Build wheels for CPython 3.10
+        run: |
+          python -m cibuildwheel --output-dir dist
+        env:
+          CIBW_BUILD: "cp310-*"
+          CIBW_MANYLINUX_X86_64_IMAGE: manylinux2010
+          CIBW_MANYLINUX_I686_IMAGE: manylinux2010
```
Comment (QuLogic) [inline .github/workflows/cibuildwheel.yml:73]: I guess there's no point to use manylinux1 for Python 3.10 now?

Comment (hrnciar) [reply .github/workflows/cibuildwheel.yml:73]: Yes, Python 3.10 does not support manylinux1 and it was failing with it.

Comment (henryiii) [reply .github/workflows/cibuildwheel.yml:73]: There's actually no point to use manylinux1 for 3.9 either - the last reason for manylinux1 was the weird hack of attaching pip 9 to `python3.8-dev` on Ubuntu 18.04. NumPy does not ship manylinux1 wheels for 3.9 anymore.

**Comment by tacaswell** [comment]
Sorry for the draft thrashing, I was optomistic that the numpy issue being closed meant they had uploaded wheels already.

**Comment by hrnciar** [comment]
@tacaswell numpy 1.21.2 was released and wheels are available. My PR works for Linux and Windows wheels, but I am getting an error on macOS build. It seems to me like an error in Numpy, WDYT? https://github.com/hrnciar/matplotlib/runs/3338097720

File: .github/workflows/cibuildwheel.yml (hunk 1 above)
Comment (hroncok) [reply .github/workflows/cibuildwheel.yml:73]: The 3.10 wheels for numpy are manylinux2014. I suppose here we should do the same?

**Comment by henryiii** [comment]
> numpy 1.21.2 was released and wheels are available. My PR works for Linux and Windows wheels, but I am getting an error on macOS build.

Only 64-bit linux (Intel and ARM) wheels are included. There are no Windows or macOS wheels. NumPy on macOS requires special setup because it can't use the buggy and deprecated built-in Accelerate framework.

**Comment by rgommers** [comment]
> NumPy on macOS requires special setup because it can't use the buggy and deprecated built-in Accelerate framework.

Not quite true anymore, Accelerate was re-enabled in `numpy.distutils` recently after some work by Apple. But we still use OpenBLAS for wheels. And it will not be re-enabled for SciPy.

**Comment by QuLogic** [comment]
I think this will need a rebase.

But while this is likely to pass a build, it does not mean that tests are passing, due to problems in our dependencies, see https://github.com/matplotlib/matplotlib/pull/20864#issuecomment-902480508

**Comment by tacaswell** [comment]
Put on the 3.5.0 milestone so we get auto backporting when this is merged.

**Comment by QuLogic** [comment]
I skipped macOS and bumped 3.10 wheels to manylinux2014: https://github.com/QuLogic/matplotlib/runs/3489831873 There don't appear to be 32-bit wheels for NumPy, so those still get rebuilt, so not sure if that saved anything.

**Comment by hroncok** [comment]
Let's skip 32bit wheels as well? They can always be added later.

**Comment by QuLogic** [comment]
I mean, they build, just a bit slower; no need to skip them.

**Review by tacaswell — APPROVED** [review]

**Comment by tacaswell** [comment]
@QuLogic if you think this is ready to go please self-merge.

**Pull request merged**